compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -377,6 +377,12 @@ object Checking {
         fail(CannotHaveSameNameAs(sym, cls, CannotHaveSameNameAs.CannotBeOverridden))
         sym.setFlag(Private) // break the overriding relationship by making sym Private
       }
+    checkApplicable(UnusedType, !sym.is(UnusedType))
+    if (sym.is(Unused)) {
+      checkApplicable(Unused, !sym.is(MutableOrLazy))
+      if (sym.info.widen.finalResultType.isBottomType)
+        fail("unused " + sym.showKind + " cannot have type Nothing")
```

Comment:
It was because of #3340 when I still was lifting the args. The program behaved differently depending on which arguments were lifted. 

I should be able to remove this now.